docs/handlers.md

```diff
@@ -0,0 +1,118 @@
+site. This is intentional on the part of Debugr: this way we can _guarantee_ that any mutable
+data passed to Debugr will not be altered by user-land code before handlers get a chance
+to process the data, or take a snapshot to be processed later. Consider the following:
+
+```typescript
+const user = await em.find(User, 'some id');
+
+logger.info('Resolved user:', { user }); // 1
+
+user.name = request.newName;             // 2
+await em.persist(user);                  // 3
+
+logger.info(['Renamed to "%s"', user.name]);
+```
+
+The `logger.info()` call at `1` will synchronously delegate to `logger.log()`, which
+will also synchronously delegate to `logger.add()`, which, in turn, will synchronously
+delegate to all registered handlers' `handler.log()`. Now imagine a Handler would perform
+an asynchronous operation inside its `log()` before processing the passed log entry.
+The execution of that handler's `log()` method would pause at the point of the first `await`
+and the rest of the _synchronous_ code which was executing up to that point would continue
+executing - so any remaining handlers' `log()` would be called, then `logger.add()` would
+return, then `logger.log()` would return, then `logger.info()` would return and then the
+line marked `2` would be executed, altering the `User` object, before execution of the original
+synchronous code was paused at the `await` on line `3` (well, more precisely, somewhere
+_within_ the `em.persist()` call). Only afterwards could the asynchronous code in the
+handler's `log()` be resumed - but the `User` object contained in the log entry the handler
+is still processing has already been mutated, and so the handler will incorrectly log
+something roughly equivalent to `Resolved user: { name: "${request.newName}" }`.
+
+So how do you avoid this? Well, if your handler serialises the entry in some way, e.g. to
+send it over the wire to an external service, then the only thing you need to do is make
+sure this serialisation takes place during the synchronous part of code execution, that is,
+before the code stops to `await` something. But also note that when you `await someFunctionCall()`,
+then the synchronous code _within_ `someFunctionCall()` also gets executed synchronously
+at the point the function is called, and only when an `await` is encountered is execution
+paused and a Promise is returned to be awaited on the previous level. Thus, for example,
+the [Elastic handler] doesn't suffer from this issue, even though it doesn't explicitly
+do anything to avoid it - because its `log()` method synchronously delegates to the Elastic
+SDK's `index()` method, which, also synchronously, serialises the entire request payload
+using `JSON.stringify()`. So there is no chance of user-land code possibly modifying the
+entry data between the data entering Debugr internals and being serialised to be sent
+to Elastic.
+
+If your code _doesn't_ serialise the entry, your other option is to use the `clone()`
```

Comment:
A co kdybychom to proste serializovali my a problem solved?

Reply:
nope, zbytecnej performance overhead, we've talked about this

Reply:
1x serializace? Neudela tolik... A usetrime si celou tuto nejasnost.

Reply:
podstatny je, ze ta serializace by se dela _u kazdy entry_; vzpomen si, ze treba u HTTP requestu a response umime logovat cely `body`, u SQL logujeme parametry (coz muzou bejt klidne i celkem hluboko nested objekty kdyz ukladas nejakej `json` treba) - to muze vytvorit celkem solidni performance penalty, a pritom to v drtivy vetsine pripadu nebude potreba (pokud si to bud ohandli primo handlery, nebo si poucenej user ten jeden pripad kdy by to delalo problemy vyresi na urovni app).. za me je to dost jednoznacny..

Reply:
jako kazdopadne by o tom melo bejt info i v end-user dokumentaci

Reply:
pridano do hlavniho `README.md`, mrkni, mozna to jde vyresit takhle (a pak by to handlery nemusely resit vubec)